backend/dataall/core/environment/services/environment_service.py

```diff
@@ -459,6 +459,32 @@ def remove_consumption_role(session, uri, env_uri):
         )
         return True
 
+    @staticmethod
+    @has_tenant_permission(permissions.MANAGE_ENVIRONMENTS)
+    @has_resource_permission(permissions.REMOVE_ENVIRONMENT_CONSUMPTION_ROLE)
+    def update_consumption_role(session, uri, env_uri, input):
+        role_query = session.query(ConsumptionRole).filter(
+            (
+                and_(
+                    ConsumptionRole.consumptionRoleUri == uri,
+                    ConsumptionRole.environmentUri == env_uri,
```

Comment:
The consumptionRoleUri is unique, we do not need the environmentUri filter, which means that in the mutation we can just have `consumptionUri` and `updateConsumptionUriInput` as inputs

Reply:
Make sense, I will remove it from the entire stack

Reply:
actually are we sure it's unique? [get_environment_consumption_role](https://github.com/data-dot-all/dataall/blob/823e8b11b483a0eb5d8d80f4cb665355a95eb56b/backend/dataall/core/environment/services/environment_service.py#L843) doesn't make that assumption? I think I will reuse this method as well (I should have anyway) for safety

Reply:
good call - let's re-use that method `get_environment_consumption_role`